### PR TITLE
Add GtkListBox example with model, property bindings and a custom row…

### PR DESCRIPTION
**Pull request by sdroege**
… GObject type

### CODE REVIEW

**Comment by EPashkin** [comment]
@sdroege Big thanks, 👍

File: src/bin/listbox_model.rs [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+            dialog.connect_response(|dialog, _| dialog.destroy());
+
+            let content_area = dialog.get_content_area();
+
+            let entry = gtk::Entry::new();
+            item.bind_property("name", &entry, "text")
+                .flags(glib::BindingFlags::DEFAULT | glib::BindingFlags::SYNC_CREATE | glib::BindingFlags::BIDIRECTIONAL)
+                .build();
+            entry.connect_activate(clone!(dialog => move |_| {
+                dialog.response(0);
```
Comment (EPashkin) [inline src/bin/listbox_model.rs]: @sdroege What does this signal handler do?

Comment (sdroege) [reply src/bin/listbox_model.rs]: When you press enter inside the `GtkEntry`, it sends response `0` (see above, "Ok") to the dialog. It is as if you clicked the corresponding button on the dialog

Comment (EPashkin) [reply src/bin/listbox_model.rs]: Thanks

**Comment by sdroege** [comment]
And all green :)

**Comment by EPashkin** [comment]
cc @GuillaumeGomez

File: src/bin/listbox_model.rs [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+
+use gio::prelude::*;
+use gtk::prelude::*;
+
+use std::env::args;
+
+use row_data::RowData;
+
+// make moving clones into closures more convenient
+macro_rules! clone {
```
Comment (GuillaumeGomez) [inline src/bin/listbox_model.rs]: We should really add this into a crate or somewhere...

**Comment by GuillaumeGomez** [comment]
Can you please add a few comments along the code? It'd help readers to understand what's happening. :)

File: src/bin/listbox_model.rs (hunk 2 above)
Comment (sdroege) [reply src/bin/listbox_model.rs]: clone_army! in the closet crate :)

**Comment by sdroege** [comment]
Sure, will do :) are there any specifics parts that you would like to have explained in more details?

Apart from comments it looks good to you?

File: src/bin/listbox_model.rs (hunk 2 above)
Comment (GuillaumeGomez) [reply src/bin/listbox_model.rs]: XD

**Comment by GuillaumeGomez** [comment]
Yes, just few comments. And explain what's going on in some cases that you think could be tricky. For us, gnome users, it's hard to know which parts are difficult to understand to newcomers so I can't help you much on that. :-/

**Comment by sdroege** [comment]
Added lots of comments, no code changes

**Comment by GuillaumeGomez** [comment]
Thanks a lot! :)

**Pull request merged**